### PR TITLE
[7196] - Add POST /trainees/{trainee_id/recommend-for-qts part 1

### PR DESCRIPTION
**Pull request by Reyko**
### Context

[7196-endpoint-post-trainees-traineeid-recommend-for-qts](https://trello.com/c/BZDMISax/7196-endpoint-post-trainees-traineeid-recommend-for-qts)

### Changes proposed in this pull request

* Add endpoint to recommend trainees for qts award
* Update docs for `v0.1`, `v1.0`

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

**Review by tomtrentham — COMMENTED** [review]
This is looking good, just a couple of questions about it in the code.

Could you also add the endpoint to the Register [Hoppscotch](https://hoppscotch.io/) workspace collection so we've got it there for convenience?

File: app/services/api/trainees/award_recommendation_service.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+        @trainee = trainee
+      end
+
+      def call
+        trainee.attributes = trainee_attributes
+
+        if valid? &&
+            trainee.submission_ready? &&
+            outcome_date_form.save! &&
```
[the remark this answers is not in the document]
Comment (Reyko) [reply app/services/api/trainees/award_recommendation_service.rb:30]: I'll take a look at this 👍

File: spec/requests/api/v1_0/post_recommend_for_qts_spec.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
```
[the remark this answers is not in the document]
Comment (Reyko) [reply spec/requests/api/v1_0/post_recommend_for_qts_spec.rb:1]: Good spot! 😅

File: app/services/api/trainees/award_recommendation_service.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (Reyko) [reply app/services/api/trainees/award_recommendation_service.rb:30]: Refactored at https://github.com/DFE-Digital/register-trainee-teachers/pull/4389

File: app/controllers/api/trainees/award_recommendations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+module Api
+  module Trainees
+    class AwardRecommendationsController < Api::BaseController
+      include Api::Serializable
+
+      def create
+        success, errors = AwardRecommendationService.call(award_recommendation_params, trainee)
```
Comment (defong) [inline app/controllers/api/trainees/award_recommendations_controller.rb:9]: Feels a bit weird service pattern we adopting

File: spec/requests/api/v0_1/trainees/post_recommend_for_qts_spec.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+           headers: { authorization: "Bearer #{token}" },
+           params: { data: { qts_standards_met_date: nil } }, as: :json
+
+      expect(response).to have_http_status(:unprocessable_entity)
+      expect(trainee.recommended_for_award_at).to be_nil
+      expect(trainee.recommended_for_award?).to be(false)
+
+      expect(response.parsed_body[:errors]).to contain_exactly(
+        "error" => "UnprocessableEntity",
+        "message" => "Qts standards met date can't be blank",
```
Comment (defong) [inline spec/requests/api/v0_1/trainees/post_recommend_for_qts_spec.rb:65]:
```suggestion
        "message" => "QTS standards met date can't be blank",
```

File: spec/services/api/trainees/award_recommendation_service_spec.rb [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+          {
+            qts_standards_met_date: Time.zone.tomorrow.iso8601,
+          }
+        end
+
+        it "returns false" do
+          success, errors = subject.call(params, trainee)
+
+          expect(success).to be(false)
+          expect(errors.full_messages).to contain_exactly("Qts standards met date must be in the past")
```
Comment (defong) [inline spec/services/api/trainees/award_recommendation_service_spec.rb:110]:
```suggestion
          expect(errors.full_messages).to contain_exactly("QTS standards met date must be in the past")
```

**Review by defong — APPROVED** [review]
Just some minor questions.
Looks fine to me

File: app/controllers/api/trainees/award_recommendations_controller.rb (hunk 3 above)
Comment (Reyko) [reply app/controllers/api/trainees/award_recommendations_controller.rb:9]: Feel free to dm me with any questions or suggestions

**Pull request merged**